tests/network/libs/dhcpd.py

```diff
@@ -0,0 +1,57 @@
+import shlex
+
+from pyhelper_utils.shell import run_ssh_commands
+from timeout_sampler import TimeoutExpiredError, TimeoutSampler
+
+from utilities.constants import TIMEOUT_5SEC, TIMEOUT_30SEC
+from utilities.network import LOGGER
+from utilities.virt import VirtualMachineForTests
+
+DHCP_IP_RANGE_START = "10.200.3.3"
+DHCP_IP_RANGE_END = "10.200.3.10"
+DHCP_IP_SUBNET = "10.200.3"
+DHCP_SERVICE_RESTART = "sudo systemctl restart dhcpd"
+DHCP_SERVER_CONF_FILE = """
+cat <<EOF >> /etc/dhcp/dhcpd.conf
+default-lease-time 3600;
+max-lease-time 7200;
+authoritative;
+subnet {DHCP_IP_SUBNET}.0 netmask 255.255.255.0 {{
+  option subnet-mask 255.255.255.0;
+  option routers {DHCP_IP_SUBNET}.1;
+  option domain-name-servers {DHCP_IP_SUBNET}.1;
+
+  pool {{
+    range {DHCP_IP_RANGE_START} {DHCP_IP_RANGE_END};
+    allow known-clients;
+    deny  unknown-clients;
+  }}
+}}
+host intended_client_vm {{
+  hardware ethernet {CLIENT_MAC_ADDRESS};
+}}
+EOF
+"""
+
+
+def verify_dhcpd_activated(vm: VirtualMachineForTests) -> bool:
+    active = "active"
+    dhcpd = "dhcpd"
+    sample = None
```

Comment:
Again - I know it was this way in the original implementation, and I also may be wrong here, but I think there's no need to init `sample`. It is anyway supposed to be set when iterating over `sampler`, and has no justification to exist without `sampler`.

Reply:
I think you are right, but it should be done in a follow up PR together with other cleanups here (e.g. use the `retry` decorator over the direct sample thing and the removal of the timeout consts).

Reply:
you need to assign `sample` here, else the you will have `sample` in the except clause as "used before assigned..."